core/src/common.cpp

```diff
@@ -250,7 +250,43 @@ namespace irods::http
 	}
 	return encoded_data;
   }
-  
+
+	auto create_host_field(boost::urls::url_view _url, std::string_view _port) -> std::string {
+		if ((_port.compare("443") == 0 && _url.scheme_id() == boost::urls::scheme::https) ||
+			(_port.compare("80") == 0 && _url.scheme_id() == boost::urls::scheme::http)) {
+			return _url.host();
+		}
+		return fmt::format("{}:{}", _url.host(), _port);
+	}
+
+	auto create_oidc_request(boost::urls::url_view _url) -> beast::http::request<beast::http::string_body> {
+		constexpr auto version_number{11};
+		beast::http::request<beast::http::string_body> req{beast::http::verb::post, _url.path(), version_number};
+
+		const auto port{get_port_from_url(_url)};
+
+		req.set(beast::http::field::host, create_host_field(_url, *port));
+		req.set(beast::http::field::user_agent, irods::http::version::server_name);
+		req.set(beast::http::field::content_type,
+		        "application/x-www-form-urlencoded");
+		req.set(beast::http::field::accept, "application/json");
+
+		if (const auto& g_config{irods::http::globals::oidc_configuration()}; g_config.contains("client_secret")) {
+			const auto format_bearer_token{[](std::string_view _client_id, std::string_view _client_secret){
```

Comment:
No, but I do kinda like it since, in my opinion, it makes clear what it does 'format_bearer_token', and cleans up the initialization in the variable following the definition. Not opposed to removing, though.

Reply:
It's fine for now.